src/compiler/utilities.ts

```diff
@@ -10082,7 +10082,28 @@ export function skipTypeChecking(sourceFile: SourceFile, options: CompilerOption
     return (options.skipLibCheck && sourceFile.isDeclarationFile ||
         options.skipDefaultLibCheck && sourceFile.hasNoDefaultLib) ||
         options.noCheck ||
-        host.isSourceOfProjectReferenceRedirect(sourceFile.fileName);
+        host.isSourceOfProjectReferenceRedirect(sourceFile.fileName) ||
+        !canIncludeBindAndCheckDiagnsotics(sourceFile, options);
+}
+
+/** @internal */
+export function canIncludeBindAndCheckDiagnsotics(sourceFile: SourceFile, options: CompilerOptions) {

```

Comment:
Observation: it seems like every caller of `canIncludeBindAndCheckDiagnsotics` writes `!canIncludeBindAndCheckDiagnsotics(...)`. Perhaps it should be inverted and be `shouldExcludeBindAndCheckDiagnsotics`?

Reply:
Not all any more so i will let this be.